providers/os/connection/snapshot/volumemounter.go

```diff
@@ -102,7 +120,37 @@ func (m *VolumeMounter) getFsInfo() (*fsInfo, error) {
 	return nil, err
 }
 
-func (m *VolumeMounter) mountVolume(fsInfo *fsInfo) error {
+// GetDeviceForMounting iterates through all the partitions of the target and returns the first one that matches the filters
+// If target is empty, it will return the first unnamed block device (best-effort guessing)
+// E.g. if target is "sda", it will return the first partition of the block device "sda" that satisfies the filters
+func (m *VolumeMounter) GetDeviceForMounting(target string) (*PartitionInfo, error) {
```

Comment:
this is now super explicit in what it does. if we have a target, we only look for mountable partitions there. if the target is empty only then we do best-effort guessing.

Reply:
we can also extend this method to support different filters on the block devices

Reply:
maybe we should call this `GetTargetForMounting` 🤔

Reply:
we still have `func (m *VolumeMounter) getFsInfo() (*PartitionInfo, error) {` in the code, which is v similar to this new func - is that intentional until we can test things more?

Reply:
yep! i didnt wanna touch the old code but it should go away soon